examples/ota-provider-app/esp32/README.md

```diff
@@ -0,0 +1,73 @@
+# CHIP ESP32 OTA Provider Example
+
+A prototype application that demonstrates OTA provider capabilities.
+
+## Supported Devices
+
+-   This example supports ESP32 and ESP32C3. For details please check
+    [here](https://github.com/shubhamdp/connectedhomeip/tree/shubhamdp-patch-1/examples/all-clusters-app/esp32#supported-devices).
+
+## Building the Example Application
+
+-   If you are building for the first time please check
+    [Building the Example Application](https://github.com/shubhamdp/connectedhomeip/tree/shubhamdp-patch-1/examples/all-clusters-app/esp32#building-the-example-application)
+    guide.
+-   Otherwise, `idf.py build` works!
+
+## Flashing the Example Application
+
+```
+idf.py -p <OTAProviderSerialPort> flash
+```
+
+## Flashing the hello-world.bin OTA image
```

Comment:
Since we are using file path for generating BDX URIs, I'd sugesst to use [SPIFFS](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/storage/spiffs.html) instead of a raw partition. IDF also provides tools to create the file system and its contents at build time.